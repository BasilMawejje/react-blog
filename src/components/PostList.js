import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'

import { fetchPosts } from '../actions'

class PostList extends React.Component{
    componentDidMount() {
        this.props.fetchPosts();
    }

    // componentDidUpdate(prevProps) {
    //     const { posts } = this.props;

    //     if(prevProps.posts !== posts) {
    //         this.props.fetchPosts();
    //     }
    // }

    // componentWillReceiveProps(nextProps) {
    //     if(nextProps) {
    //         this.props.fetchPosts();
    //     }
    // }

    renderList() {
        return this.props.posts.map( post => {
            return (
                <div className="item" key={post.id}>
                    <div className="content">
                            <Link to={`/posts/${post.id}`} className='header'>
                                <h2>{post.title}</h2>
                            </Link>
                        <div className="description">
                            <p>{post.body}</p>
                        </div>
                    </div>
                    <Link to={`/posts/edit/${post.id}`} className='ui button primary'>Edit</Link>
                    <Link to={`/posts/delete/${post.id}`} className='ui button negative'>Delete</Link>
                </div>
            );
        });
    }

    renderCreate() {
        return (
            <div>
                <Link to='/posts/new' className='ui button primary'>
                    Create Post
                </Link>
            </div>
        );
    }

    render() {
        return (
                <div>
                    <h1>List of Posts</h1>
                    {this.renderCreate()}
                    <div className="ui celled list">{this.renderList()}</div>
                </div>
            );
    }
}

const mapStateToProps = state => {
    return { posts: Object.values(state.posts) }
}

export default connect(mapStateToProps, { fetchPosts })(PostList);