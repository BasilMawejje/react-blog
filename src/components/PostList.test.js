import React from 'react';
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux'
import { shallowToJson } from 'enzyme-to-json';

import PostList from './PostList'

configure({ adapter: new Adapter() });

const mockStore = configureStore();
const store = mockStore({posts: 
    [{ id: 1, title: 'Some post', body: 'lorem ipsum ...' },
    { id: 2, title: 'Sample title', body: 'sample description' }]
});

describe('PostList', () => {
    let wrapper; 

    beforeEach(() => {
        wrapper = shallow(
            <Provider store={store}>
                <PostList />
            </Provider>
        ).dive();
    });

    it('renders properly', () => { 
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('renders a list of items', () => {
        expect(wrapper.find(PostList)).toBeTruthy();
        expect(wrapper.find(PostList)).toHaveLength(1);
    });
})