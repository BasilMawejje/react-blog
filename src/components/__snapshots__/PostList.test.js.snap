// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList renders properly 1`] = `<Connect(PostList) />`;
